lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -814,4 +814,16 @@ def b = new CpsTransformerTest.Base()
 return (b.&toString)() + (String.getClass().&getSimpleName)()
 ''') == "baseClass"
     }
+
+    @Issue("JENKINS-44027")
+    @Test
+    void multipleAssignment() {
```

Comment:
Needs to have some tests for the weirder/error cases I think to ensure we're exercising the code paths properly